src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1209,10 +1208,20 @@ tx_gateway_frontend::do_end_txn(
             co_return tx_errc::invalid_txn_state;
         }
     } else {
-        r = co_await do_abort_tm_tx(stm, tx, timeout, outcome);
+        try {
+            r = co_await do_abort_tm_tx(stm, tx, timeout);
+        } catch (...) {
+            outcome->set_value(tx_errc::unknown_server_error);
+            throw;
+        }
```

Comment:
I think the only thing worth asking about here is: this try {} block covers many more LOC than the try {} that had been inside do_abort_tm_tx (e.g. `auto prs = co_await when_all_succeed(pfs.begin(), pfs.end());`).

So prior to this commit an execution might have been:

```
   outcome->set_value(tx_errc::none);
   auto prs = co_await when_all_succeed(pfs.begin(), pfs.end());
```

where `co_await` throws and is bubbled up and outcome contains no error.

now the same failure scenario would result in an outcome with unknown server error rather than `tx_errc::none`. in both cases since the exception is bubbled up its probably taking precedence some where and the outcome doesn't matter, but it did seem like a subtle difference in semantics worth asking about.

Reply:
`unknown_server_error` means that both outcomes are possible also the exceptions should be rare so it's ok to be less precise here.